### PR TITLE
Add test execution guide in yamlRestTest asciidoc

### LINKED ISSUE #88406: Need explanation of how to run tests in rest-api-spec test guide
[The rest-api-spec tests guide](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc) is confusing because after all the explanations of the test file structure, it doesn't tell us how to actually _RUN_ an individual test or a test suite.

Luckily for me, I happened to run into [this comment](https://github.com/elastic/elasticsearch/issues/83957#issuecomment-1121014066) that explains how to run such a test.

It would be better if we could add an explanation of how to run tests (or a link to some doc located elsewhere) to the guide. It won't be much work but would save a lot of time for the developers.

I'd be glad to create a PR for this.

### PR DESCRIPTION
**Pull request by chen-ni**
Fix https://github.com/elastic/elasticsearch/issues/88406

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc [hunk 1]
```diff
@@ -15,6 +15,10 @@ bin/elasticsearch -Enode.attr.testattr=test -Epath.repo=/tmp -Erepositories.url.
 
 =======================================
 
+Test execution
+---------------
+Please refer to https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#testing-the-rest-layer[this section].
```
Comment (mark-vieira) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc:20]: Let's use relative links so we aren't hard-coding any particular branch.

```suggestion
Please refer to /TESTING.asciidoc#testing-the-rest-layer[this section].
```

**Review by mark-vieira — APPROVED** [review]
One minor suggestion, otherwise 👍.

**Comment by mark-vieira** [comment]
@elasticmachine ok to test

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc (hunk 1 above)
Comment (chen-ni) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc]: @mark-vieira Thanks, fixed :)

**Comment by chen-ni** [comment]
@mark-vieira Unfortunately, the updated version doesn't seem to work. The following syntax is not recognized as a link in both IntelliJ and on GitHub:
```
/TESTING.asciidoc#testing-the-rest-layer[this section]
```

I then tried converting it into an `xref`:
```
xref:/TESTING.asciidoc#_testing_the_rest_layer[this section]
```

where `#_testing_the_rest_layer` is the auto-generated asciidoc ID for the section. But this doesn't work either, apparently because `xref` doesn't support referring document from root directory (`/`).

Now I'm stuck. I referred to the [official guide](https://docs.asciidoctor.org/asciidoc/latest/macros/inter-document-xref/) and [this SO question](https://stackoverflow.com/questions/33549013/asciidoc-cross-reference-to-a-document-in-a-directory-on-the-same-level) but they don't help much.

**Comment by chen-ni** [comment]
Just submitted https://github.com/asciidoctor/asciidoctor/issues/4313 to ask about it.

**Comment by chen-ni** [comment]
@mark-vieira `xref` works now on GitHub after turning `compat-mode` off by adding `:!compat-mode:`. (It still doesn't work in IntelliJ, but I think it doesn't matter.)

Could you please review again, thanks :)

**Comment by mark-vieira** [comment]
Thanks @chen-ni!

**Pull request merged**